src/clj/game/core/runs.clj

```diff
@@ -195,47 +211,84 @@
+        ;; strs
+        steal-str [(if can-steal-this?
+                     (if can-pay-costs?
+                       (if cost-as-symbol
+                         "Pay"
+                         "Steal")
+                       "OK")
+                     "OK")]
+        dont-pay-str (when can-steal-this?
+                       (when (and can-pay-costs?
+                                  cost-as-symbol)
+                         ["Don't pay"]))
+        prompt-str (str "You accessed " card-name "."
+                        (when can-steal-this?
+                          (if (= 1 (count choices))
+                            (str " Pay " (costs-to-symbol cost) " to steal?")
+                            " Steal?"))
+                        (when (pos? (count ability-strs))
+                          " Use an ability to trash?"))
+        ]
+    ;; Steal costs are additional costs and can be denied by the runner.
+    (continue-ability state :runner
+                      (if (> (count choices) 1)
+                        ;; Use the better function for multiple costs
+                        (steal-pay-choice state :runner choices '() n c)
+                        ;; Otherwise, just handle everything right friggin here
+                        {:delayed-completion true
+                         :prompt prompt-str
+                         :choices (into [] (concat ability-strs
+                                                   steal-str
+                                                   dont-pay-str))
+                         :effect (req (cond
+                                        ;; Can't steal
+                                        (= target "OK")
+                                        (do (trigger-event state side :no-steal c)
+                                            (resolve-steal-events state side eid c))
+
+                                        ;; Steal normally
+                                        (= target "Steal")
+                                        (resolve-steal state :runner eid c)
+
+                                        ;; Pay single additiional cost to steal
+                                        (= target "Pay")
+                                        (when-completed (pay-sync state side nil cost {:action :steal-cost})
+                                                        (do (system-msg
+                                                              state side
+                                                              (str "pays " cost-as-symbol " to steal " card-name))
+                                                            (resolve-steal state side eid c)))
+
+                                        ;; Don't pay single additional cost to steal
+                                        (.contains target "Don't")
+                                        (do (trigger-event state side :no-steal c)
+                                            (effect-completed state side eid))
```

Comment:
I think we want to call `access-end` here instead of `effect-completed`.

Reply:
Oh whoops, good call.